main.go

```diff
@@ -140,6 +176,100 @@ func main() {
 	}
 }
 
+func deleteAction(ctx context.Context, cancel context.CancelFunc, clientSet client.Client, hr *fluxhelmv2beta1.HelmRelease, timeout, interval time.Duration) {
+	instance := &fluxhelmv2beta1.HelmRelease{}
+	key := client.ObjectKey{
+		Name:      hr.Name,
+		Namespace: hr.Namespace,
+	}
+	if err := clientSet.Get(ctx, key, instance); client.IgnoreNotFound(err) != nil {
+		log.Fatalf("Failed to get instance of the helmrelease with %v", err)
+	} else if err != nil {
+		// Unexpectedly, the object that we are trying to delete was not found
+		// If this happens, we will log an error and return
+		log.Errorf("Did not find the helm release: %v", hr.Name)
+		cancel()
+		return
+	} else {
+		instance.Annotations = hr.Annotations
+		instance.Labels = hr.Labels
+		instance.Spec = hr.Spec
+
+		log.Infof("Found the helm release: %v, deleting the release...", hr.Name)
+		// Delete the HelmRelease
+		if err := retry(ctx, func() error { return clientSet.Delete(ctx, instance) }, interval); err != nil {
+			log.Fatalf("retry got err: %v", err)
+		}
+	}
+
+	pollingFunc := func() bool {
+		instance := &fluxhelmv2beta1.HelmRelease{}
+		key := client.ObjectKey{
+			Name:      hr.Name,
+			Namespace: hr.Namespace,
+		}
+		if err := clientSet.Get(ctx, key, instance); client.IgnoreNotFound(err) != nil {
+			if ctx.Err() == context.DeadlineExceeded {
```

Comment:
This may be unnecessary, we will get the timeout log twice in this case, once from this function and the other from the polling func